src/lib/libast/misc/fallbacks.c

```diff
@@ -22,6 +25,65 @@ void do_not_use_this_fallback() {
     abort();
 }
 
+#if !_lib_eaccess
+#if _lib_euidaccess
+// System doesn't have eaccess() but does have the equivalent euidaccess() so use that.
+int eaccess(const char *pathname, int mode) {
+    return euidaccess(pathname, mode);
+}
+#elif _lib_faccessat
+// System doesn't have eaccess() but does have faccessat() so use that.
+int eaccess(const char *pathname, int mode) {
+    return faccessat(AT_FDCWD, pathname, mode, AT_EACCESS);
+}
+#else
+// The platform doesn't have eaccess(), euidaccess(), or faccessat() so we have to roll our own.
+// This may not be optimal in that it calls access() when it might be possible to return an answer
+// based on what we already know. Since this is a fallback function we hope not to use we're
+// striving for simplicity and clarity.
+int eaccess(const char *pathname, int mode) {
+    // In case the user passed bogus bits for the mode tell them we can't determine what they want.
+    if (mode & ~(X_OK | W_OK | R_OK)) {
```

Comment:
Add `F_OK` to this list. It's set to `0` on my Linux system, but unless you are sure that it's always `0`, it should be here.

Other option is to remove bogus bits like [glibc implementation](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/euidaccess.c;h=ec44b0f6961edcb19b8bf67df3be530437b4902c;hb=HEAD#l146).

Reply:
I would expect it to be zero on all UNIX like systems but I'll include it here rather than remove bogus bits. If the user passes a bogus value it isn't safe to assume the what is left after stripping the invalid bits is meaningful.